Common/Logger.swift

```diff
@@ -119,29 +157,72 @@ class Logger {
-            zLog.error("Unable to capture stdout and stderr to file \(url.path)")
-            return false
+            zLog.error("Unable to capture stdout and stderr to file \(currLog.path)")
         }
         setbuf(__stdoutp, nil) // set stdout to flush
         
+        // rotate logs (potentially))
+        if let attrs = try? fm.attributesOfItem(atPath: currLog.path) as NSDictionary {
+            let sz = attrs.fileSize()
+            
+            var dailyRotateNeeded = false
+            if rotateDaily, let lrt = lastRotateTime {
+                if !Calendar.current.isDate(lrt, inSameDayAs:Date()) {
+                    dailyRotateNeeded = true
+                }
+            }
+            
+            if force || dailyRotateNeeded || (sz >= Logger.FILE_SIZE_THRESHOLD) {
+                for i in (1...(Logger.MAX_NUM_LOGS-2)).reversed() {
+                    let from = currLog.appendingPathExtension("\(i)")
+                    let to = currLog.appendingPathExtension("\(i+1)")
+                    
+                    do {
+                        try? fm.removeItem(at: to)
+                        try fm.moveItem(at: from, to: to)
+                    } catch {
+                        zLog.error("Error rotating \(from.path) to \(to.path): \(error.localizedDescription)")
+                    }
+                }
+                
+                // roll out the current log
+                let to = currLog.appendingPathExtension("1")
+                do {
+                    try? fm.removeItem(at: to)
+                    
+                    // copy instead of mv since we want to manipulate file handle to move to start of the file
+                    // to preserve any running `tail -f` on current log
+                    try fm.copyItem(at: currLog, to: to)
+                    try lfh.truncate(atOffset: 0)
+                    try lfh.seek(toOffset: 0)
+                    
+                    // Start each log by logging app version
+                    zLog.info(Version.verboseStr)
+                } catch {
+                    zLog.error("Error rotating \(currLog.path) to \(to.path): \(error.localizedDescription)")
+                }
+                
+                // update lastRotateTime
+                lastRotateTime = Date()
+            }
+        }
         return true
     }
     
     static func initShared(_ tag:String) {
         Logger.shared = Logger(tag)
+        Logger.shared?.initLogDir()
+        
         zLog.info("Setting log level to \(ZitiLog.LogLevel.INFO)")
         ZitiLog.setLogLevel(.INFO)
-        if Logger.shared?.updateLogger() == false {
-            // Do what?  invalidate Logger? revisit when add #file #function #line stuff
-            zLog.error("Unable to created shared Logger")
-        }
-
+        
+        // Process once at startup to make sure we have log dir, roll logs from previos runs if necessary
+        _ = Logger.shared?.rotateLogs(false)
```

Comment:
are you actually "making sure" if you discarding the response? I'll have to see if rotateLogs exits. otherwise the comment is not really in alignment withthe code imo